src/components/Graph.react.js

```diff
@@ -271,6 +279,12 @@ const graphPropTypes = {
      */
     relayoutData: PropTypes.object,
 
+    /**
+     * Data from latest restyle event which occurs
+     * when the user toggles a legend item
+     */
+    restyleData: PropTypes.object,
```

Comment:
Can we have a shape instead?

Reply:
Oh right, it’s actually a length-2 array of the args to `Plotly.restyle` - the first one an object of `<attribute.string>: <value>`, the second an integer or array of integers for the trace indices. So I’ll change it to array and describe this (yes, it’s also read-only) but it doesn’t look to me as though `PropTypes` allows us to do better other than via custom function, which wouldn’t do the user any good. Is that right or am I missing how to do that?

Reply:
Changed to `array` and improved docs (including read-only on all the event data props) in https://github.com/plotly/dash-core-components/pull/483/commits/688a66f6c7ed59aea77321a37b5f6d7e180c7304